docs/admin/kubeadm.md

```diff
@@ -525,6 +525,57 @@ Here `v1.7.x` means the "latest patch release of the v1.7 branch".
 
 `${ARCH}` can be one of: `amd64`, `arm`, `arm64`, `ppc64le` or `s390x`.
 
+## Managing the kubeadm dropin file for the kubelet
+
+The kubeadm deb package ships with configuration for how the kubelet should
+be run. Note that the `kubeadm` CLI command will never touch this dropin file.
+This dropin file belongs to the kubeadm deb/rpm package.
+
+This is what it looks like in v1.7:
+
+```
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--kubeconfig=/etc/kubernetes/kubelet.conf --require-kubeconfig=true"
+Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true"
+Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
+Environment="KUBELET_DNS_ARGS=--cluster-dns=10.96.0.10 --cluster-domain=cluster.local"
+Environment="KUBELET_AUTHZ_ARGS=--authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt"
+Environment="KUBELET_CADVISOR_ARGS=--cadvisor-port=0"
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_EXTRA_ARGS
+```
+
+A breakdown of what/why:
+ - `--kubeconfig=/etc/kubernetes/kubelet.conf` point to the kubeconfig file that
+   tells the kubelet where the API server is. This file also has the kubelet's
+   credentials.
+ - `--require-kubeconfig=true` the kubelet should fail fast if the kubeconfig file
+   is not present. This makes the kubelet crashloop during the time the service is
+   started until `kubeadm init` is run.
+ - `--pod-manifest-path=/etc/kubernetes/manifests` specifies from where to read
+   Static Pod manifests used for spinning up the control plane
+ - `--allow-privileged=true` allow this kubelet to run privileged Pods
+ - `--network-plugin=cni` use CNI networking
+ - `--cni-conf-dir=/etc/cni/net.d` where to look for the
+   [CNI spec file(s)](https://github.com/containernetworking/cni/blob/master/SPEC.md)
+ - `--cni-bin-dir=/opt/cni/bin` look for the actual CNI binaries in this directory
+ - `--cluster-dns=10.96.0.10` use this cluster-internal DNS server for `nameserver`
+   entries in Pods' `/etc/resolv.conf`
+ - `--cluster-domain=cluster.local` use this cluster-internal DNS domain for
+   `search` entries in Pods' `/etc/resolv.conf`
+ - `--client-ca-file=/etc/kubernetes/pki/ca.crt` authenticate requests to the Kubelet
+   API using this CA certificate.
+ - `--authorization-mode=Webhook` authorize requests to the Kubelet API by `POST`-ing
+   a `SubjectAccessReview` to the API Server
+ - `--cadvisor-port=0` disable that cAdvisor listens to `0.0.0.0:4194` by default.
+   cAdvisor will still be run inside of the kubelet and its API can be accessed at
+   `https://{node-ip}:10250/stats/`. If you want to enable that cAdvisor listens on
+   a wide-open port, run:
+   ```
```

Comment:
Add a newline above this line so that the shell snippet is rendered in a separate block on its own line, see preview: https://deploy-preview-4229--kubernetes-io-master-staging.netlify.com/docs/admin/kubeadm/